### PR TITLE
Closing Issue #1373

### PR DESCRIPTION
**Pull request by mzitnik**
## Summary
Will close Issue #1373 
Fix the case while we have an error in addBatch method 

## Checklist
Delete items not relevant to your PR:
- [X] Unit and integration tests covering the common scenarios were added
- [ ] A human-readable description of the changes was provided to include in CHANGELOG
- [ ] For significant changes, documentation in https://github.com/ClickHouse/clickhouse-docs was updated with further explanations or tutorials

### CODE REVIEW

**Comment by mshustov** [comment]
Should it close https://github.com/ClickHouse/clickhouse-java/issues/1373 ? Could you add a description to the PR title?

**Comment by mzitnik** [comment]
> Should it close #1373 ? Could you add a description to the PR title?

Yes it should adding in the description

File: clickhouse-jdbc/src/main/java/com/clickhouse/jdbc/internal/InputBasedPreparedStatement.java [hunk 1]
```diff
@@ -326,6 +326,7 @@ public void addBatch() throws SQLException {
 
         ClickHouseDataProcessor processor = getDataProcessor(stream, null, columns);
         int nullAsDefault = getNullAsDefault();
+        // validate the values before the actual write into the processor
```
Comment (mshustov) [inline clickhouse-jdbc/src/main/java/com/clickhouse/jdbc/internal/InputBasedPreparedStatement.java:329]: nit: Can we move it into a separate method to make the comment unnecessary?

File: clickhouse-jdbc/src/test/java/com/clickhouse/jdbc/JdbcIssuesTest.java [hunk 2]
```diff
@@ -123,5 +123,37 @@ public void test03Decompress() throws SQLException {
...
+        int count = 1;
+        boolean failed = false;
+        try (PreparedStatement ps = conn.prepareStatement(sql)) {
+            while (count <= 10) {
+                try {
+                    ps.setString(1, "******");
+                    ps.setInt(2, 10);
+                    ps.setString( 3, count == 2 ? null : "--------");
+                    ps.addBatch();
+                } catch (Exception e) {
```
Comment (mshustov) [inline clickhouse-jdbc/src/test/java/com/clickhouse/jdbc/JdbcIssuesTest.java:147]: Can the `catch` clause be removed?

**Review by mshustov — APPROVED** [review]

**Pull request merged**